### PR TITLE
fix: ENT-4865 Reduced object size of courses indexed with programs

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
## Description

This PR reduces the object size of courses indexed with programs by indexing only the course keys. This is required to meet the maximum object size requirement in Algolia.

## Ticket Link

Link to the associated ticket
[ENT-4865](https://openedx.atlassian.net/browse/ENT-4865)

### CODE REVIEW

File: enterprise_catalog/apps/catalog/algolia_utils.py [hunk 1]
```diff
@@ -322,6 +321,37 @@ def _get_course_program_field(course, field):
...
+    """
+    Helper to pluck a list of values for the given field out of a program's courses.
+
+    Arguments:
+        program (dict): a dictionary representing a program
+        field (str): the name of a field to return values of.
+    Returns:
+        list: a list of the values for a certain field in a course associated with the program.
+    """
+    programs = program.get('courses') or []
```
Comment (sameenfatima78) [inline enterprise_catalog/apps/catalog/algolia_utils.py]: nit: We can also pass a default value `[]` to `.get`

Comment (sameenfatima78) [reply enterprise_catalog/apps/catalog/algolia_utils.py]: nit: rename `programs` to `courses`

File: enterprise_catalog/apps/catalog/algolia_utils.py [hunk 2]
```diff
@@ -322,6 +321,36 @@ def _get_course_program_field(course, field):
...
+    return list({
+        value for program in programs
+        if (value := program.get(field))
+    })
```
Comment (sameenfatima78) [inline enterprise_catalog/apps/catalog/algolia_utils.py]: What will be returned if a program has no courses?

**Review by sameenfatima78 — APPROVED** [review]
Looks good, just a few comments

File: enterprise_catalog/apps/catalog/algolia_utils.py (hunk 2 above)
Comment (irfanuddinahmad) [reply enterprise_catalog/apps/catalog/algolia_utils.py]: empty list will be returned

**Pull request merged**